src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const { type } = action;
+    let nextState;
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = action;
+
+        nextState = {
+          ...currentState, ...extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = action;
```

Comment:
Same issue as above. The variable 'keysToRemove' is declared inside a switch case without block braces.